framework/wazuh/core/tests/test_agent.py

```diff
@@ -696,6 +696,7 @@ def test_agent_add_ko(mock_maganer_status):
 
 @pytest.mark.parametrize("name, ip, id, key, force", [
     ('test_agent', '172.19.0.100', None, None, None),
+    ('test_agent', 'any', '001', None, None),
```

Comment:
I think we need a test case for the combination `id=None` and `key=<some-value>`

Reply:
Agreed, I will add that as well